### PR TITLE
feat: export metrics 

### PR DESCRIPTION
**Pull request by lightsofapollo**
Adds metrics for common error cases (timeoout, dtls handshake failure) and few success cases (stream open / stream close)

Closes: https://github.com/libp2p/js-libp2p-webrtc/issues/64

### CODE REVIEW

File: test/transport.browser.spec.ts [hunk 1]
```diff
@@ -16,10 +19,51 @@ function ignoredDialOption (): CreateListenerOptions {
 
 describe('WebRTC Transport', () => {
   let components: underTest.WebRTCTransportComponents
+  let metrics: Metrics
 
   before(async () => {
+    metrics = stubObject<Metrics>({
```
Comment (lightsofapollo) [inline test/transport.browser.spec.ts:25]: I wanted to call out that this piece is arguably not super useful with the tests in their current state.  I added this here for completeness' sake. If we wish to add additional metrics or add full e2e testing, we can test that the metrics are invoked further.

**Comment by lightsofapollo** [comment]
@achingbrain David DiMaria pointed out to me that we already have this : https://github.com/libp2p/js-libp2p-interfaces/blob/master/packages/interface-mocks/src/metrics.ts . I believe sinon-ts was suggested in another comment but using the existing mock would definitely work. Do you have a preference?

**Comment by achingbrain** [comment]
They are intended to fulfil different roles - the classes in `@libp2p/interface-mocks` are functioning implementations of the various interfaces (e.g. they pass their associated interface tests), useful for when stubbing large amounts of functionality would make the test overly verbose or would introduce lots of redundant code.

Previously we had complicated dependency trees where libp2p components would depend on each other (and libp2p itself) to test functionality, the collection of mocks was intended to alleviate some of that.

You'd use `sinon-ts` when you want to assert simple interactions with the other components, like a method being invoked with certain arguments being passed, etc which seemed like a better fit here.

**Comment by ddimaria** [comment]
@achingbrain This PR collects dialer events.  I know we'll need to add listener ones when b2b is completed.  Can you think of other relevant events to include?  (`tcp` had a server status that isn't applicable here).

**Comment by BigLep** [comment]
@achingbrain : I want to get this one out of limbo.  Can you please give any input?

**Comment by p-shahi** [comment]
@achingbrain @lightsofapollo What are the next steps to get this landed?

**Comment by ddimaria** [comment]
> @achingbrain @lightsofapollo What are the next steps to get this landed?

@lightsofapollo it looks like there are conflicts that need to be resolved, and an issue with CI.  I have an outstanding [ question](https://github.com/libp2p/js-libp2p-webrtc/pull/71#issuecomment-1377689687) to @achingbrain, though maybe @MarcoPolo can answer it?

**Comment by p-shahi** [comment]
I'll bring this up for discussion during the js-libp2p triage to determine next steps

**Comment by ddimaria** [comment]
> I'll bring this up for discussion during the js-libp2p triage to determine next steps

@p-shahi were you able to determine the next steps?

**Comment by p-shahi** [comment]
> > I'll bring this up for discussion during the js-libp2p triage to determine next steps
> 
> @p-shahi were you able to determine the next steps?

Yes, @wemeetagain is assigned for review, so waiting on that.

**Review by maschad — APPROVED** [review]
LGTM

**Review by maschad — APPROVED** [review]

**Comment by maschad** [comment]
It seems the [failing test](https://github.com/libp2p/js-libp2p-webrtc/actions/runs/4433436526/jobs/7778466472?pr=71) has been [resolved](https://github.com/libp2p/js-libp2p-webrtc/actions/runs/4372820873/jobs/7650215969?pr=90) on the [Browser-to-server PR](https://github.com/libp2p/js-libp2p-webrtc/pull/90) ,  regardless there will be merge conflicts so let's wait till that is merged first given that's of a higher priority.

**Review by achingbrain — APPROVED** [review]
LGTM, just a few small nits

**Comment by maschad** [comment]
https://github.com/libp2p/js-libp2p-webrtc/pull/150 has the necessary fixes to stop CI from failing, let's merge that first.

**Pull request merged**